File: docs/snippets/csharp/round-trip-fidelity_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;

// Original message
var original = new UserMessage
{
    MessageId = "msg-roundtrip",
    Contents = new List<AIContent>
    {
        new TextContent { Text = "Test message" }
    }
};

// Serialize then deserialize
var serializer = new MessageSerializer();
var xml = serializer.Serialize(original);
var restored = serializer.Deserialize(xml);

// Verify fidelity
Assert.Equal(original.Role, restored.Role);
Assert.Equal(original.MessageId, restored.MessageId);
Assert.Equal((original.Contents[0] as TextContent)?.Text, (restored.Contents[0] as TextContent)?.Text);

Console.WriteLine("✓ Round-trip successful");

File: docs/snippets/python/round-trip-fidelity_main.py
# from microsoft.agents.xml.models import ChatMessage, TextContent
# from microsoft.agents.xml.serialization import MessageSerializer
#
# # Original message
# original = ChatMessage(
#     role="user",
#     message_id="msg-roundtrip",
#     contents=[TextContent(text="Test message")]
# )
#
# # Serialize then deserialize
# serializer = MessageSerializer()
# xml = serializer.serialize(original)
# restored = serializer.deserialize(xml)
#
# # Verify fidelity
# assert restored.role == original.role
# assert restored.message_id == original.message_id
# assert restored.contents[0].text == original.contents[0].text
#
# print("✓ Round-trip successful")